### PR TITLE
Migrate to Astro, add Docs section

### PR DESCRIPTION
**Pull request by husseinalhammad**
Migrating to [Astro🚀](https://astro.build/). Using their [Docs Starter Kit](https://github.com/withastro/astro/tree/main/examples/docs) which is SEO friendly and has a lot of features releavant to us.

**Home page changes**
- Add Docs links to main navbar
- Replace quick links section with a CTA directing to the docs

**Home page vs Docs styles**
The home page still contains the original CSS. So it has different theming options and doesn't have a dark theme like the docs. We can address this at a later stage. The page can still act as a landing page.

The Docs default theme colours use blue accents. They're not too prominent though and we can update these. 

**Docs management**
- Markdown files (so no changes to original goal)
- Sidebar links are customisable in `src/config.ts`. We can create sections for different projects (or just to group related pages)
- Docs in [email-markup-consortium](https://github.com/email-markup-consortium/email-markup-consortium) repo have been copied over

**Deployment**
We can continue using GitHub Pages. Deploy Action `deploy.yml` has been added.

### CODE REVIEW

**Comment by husseinalhammad** [comment]
I forgot to mention that we have a Netlify account (pro plan). So we can switch to Netlify if preferred.

**Comment by husseinalhammad** [comment]
Added this repo to our Netlify account. The deploy preview will hopefully make it easier for reviewers to preview changes in the browser (as opposed to building locally).

Before merging this to the main branch, we'll need to change DNS records and update the domains settings on Netlify.

File: src/styles/index.css [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+
+table {
+	width: 100%;
+	padding: var(--padding-block) 0;
+	margin: 0;
+	border-collapse: collapse;
+}
+
+/* Zebra striping */
+tr:nth-of-type(odd) {
```
Comment (M-J-Robbins) [inline src/styles/index.css:239]: To help separate the `<thead>` from the `<tbody>` could be good to modify this to `tbody tr:nth-of-type(odd)` so the colours don't run into each other.

And maybe add a small border too `thead {border-bottom: 0.1em solid;}`

**Review by M-J-Robbins — COMMENTED** [review]
I've added a couple of notes but otherwise looking really good 👍

File: src/components/Header/EmcLogo.astro [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+const { width } = Astro.props;
+---
+
+<svg width={width} viewBox="0 0 999 281">
```
Comment (M-J-Robbins) [inline src/components/Header/EmcLogo.astro:5]: Add alt text
  
  ```
  <svg role="img">
      <title>Email Markup Consortium</title>
 ```

Comment (M-J-Robbins) [reply src/components/Header/EmcLogo.astro:5]: Looks like I didn't hit send on that second comment before submitting the review

File: src/styles/index.css (hunk 1 above)
Comment (husseinalhammad) [reply src/styles/index.css:239]: If I understand you correctly, you are saying `tbody tr:first-child` should not have the same background colour as `thead tr`? Do you think they should alternate? Or give `thead tr` a different background colour altogether?

File: src/components/Header/EmcLogo.astro (hunk 2 above)
Comment (husseinalhammad) [reply src/components/Header/EmcLogo.astro:5]: Good call. Added.

Comment (M-J-Robbins) [reply src/components/Header/EmcLogo.astro:5]: Sorry I wasn't very clear in the first message, i would also want a `role="img"` on the `<svg>` tag

File: src/styles/index.css (hunk 1 above)
Comment (M-J-Robbins) [reply src/styles/index.css:239]: Yeah I'm happy with either solution. Currently the `thead tr` is the same colour as the first row of the `tbody` so looks like they are one.

glossary page illustrating the issue

So either alternate back to the lighter colour and maybe add a `border-bottom` or add a new colour. Just so it's clearly a heading

Comment (husseinalhammad) [reply src/styles/index.css]: Makes sense. Updated

**Review by M-J-Robbins — APPROVED** [review]
all looking good

**Pull request merged**